src/site/xhp/ui-feedback.php

```diff
@@ -0,0 +1,104 @@
+<?hh // strict
+/*
+ *  Copyright (c) 2004-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+namespace HHVM\UserDocumentation\ui;
+
+use namespace Facebook\XHP\Core as x;
+use type Facebook\XHP\HTML\{a, div, i, script};
+use type HHVM\UserDocumentation\{
+  github_issue_link,
+};
+
+final xhp class feedback extends x\element {
+
+  <<__Override>>
+  protected async function renderAsync(): Awaitable<x\node> {
+    $thumbs_up =
+      <a id="like-feedback" href="#"><i class="fa fa-solid fa-thumbs-up fa-lg"></i></a>;
+
+    $thumbs_down =
+      <a id="dislike-feedback" href="#"><i class="fa fa-solid fa-thumbs-down fa-lg"></i></a>;
+
+    $feedback_message =
+      <div id="feedback-prompt-message">Was This Page Useful?
+        {$thumbs_up}
+        {$thumbs_down}
+      </div>;
+
+    $good_feedback_message =
+      <div id="like-message">Thank You!</div>;
+
+    $bad_feedback_message =
+      <div id="dislike-message">Thank You. If you have feedback, you can <github_issue_link issueTitle={$this->getGithubIssueTitle()} issueBody={$this->getGithubIssueBody()}>File an Issue</github_issue_link> to help future readers!</div>;
+
+    $container = (
+      <div id="feedback-container" class="pageFeedback">
+        {$feedback_message}
+        {$good_feedback_message}
+        {$bad_feedback_message}
+      </div>
+    );
+
+    $container->appendChild(varray[
+      $this->toggleFeedbackScript(),
+    ]);
+
+    return $container;
+  }
+
+  private function toggleFeedbackScript(): script {
+    return (
+      <script language="javascript">
+        var likeIcon = document.getElementById("like-feedback");
+        var dislikeIcon = document.getElementById("dislike-feedback");
+        likeIcon.addEventListener(
+        'click',
+        function(e) {"{"}
+        e.preventDefault();
+        var feedbackContainer = document.getElementById("feedback-container");
+        feedbackContainer.classList.toggle('feedbackPromptOff');
+        feedbackContainer.classList.toggle('likeFeedbackOn');
+        ga('send', 'event', 'Feedback', 'Useful', 'Guides Page', 1);
+        {"}"}
+        );
+        dislikeIcon.addEventListener(
+        'click',
+        function(e) {"{"}
+        e.preventDefault();
+        var feedbackContainer = document.getElementById("feedback-container");
+        feedbackContainer.classList.toggle('feedbackPromptOff');
+        feedbackContainer.classList.toggle('dislikeFeedbackOn');
+        ga('send', 'event', 'Feedback', 'Not Useful', 'Guides Page', 0);
+        {"}"}
+        );
```

Comment:
perhaps:

```
<script>{
<<<EOF
indented_js_here()
if (foo) {
  fewfewafa;
}
EOF
}</script>
```

This is also borderline worth splitting out to a separate file in `public/`

Reply:
it's already hard-coded in 2 other places (now 3), so good idea.

Reply:
I split it into a separate .js file in the latest commit but I can't get it to execute. Wondering if it's because it's being executed through GitHub Codespaces? Even though I set the type as 'application/javascript' it comes through as 'application/octet-stream'

<img width="1431" alt="image" src="https://user-images.githubusercontent.com/5179225/156020106-a9602aa6-df6d-4ada-a8db-13d7cec45e1f.png">

Reply:
> Wondering if it's because it's being executed through GitHub Codespaces? 

This would be pretty reasonable from a security perspective; what do you get when you use `curl` (and `curl -I`) to fetch it from codespaces?

Reply:
from terminal in codespaces I can get the content just fine.

root@codespaces_00e652:/workspaces/user-documentation# curl -I https://raw.githubusercontent.com/AndrewDiMola/user-documentation/a194d5fa17e9b1595479c055fa27c024d/public/js/feedback.js
HTTP/2 200 
cache-control: max-age=300
content-security-policy: default-src 'none'; style-src 'unsafe-inline'; sandbox
content-type: text/plain; charset=utf-8
etag: "898d36d16431b91704215f943b3bc124f822f2753a8de3b4a9c543c225fa3773"
strict-transport-security: max-age=31536000
x-content-type-options: nosniff
x-frame-options: deny
x-xss-protection: 1; mode=block
x-github-request-id: 9078:2308:9E1D5C:C47FA4:621D8A68
accept-ranges: bytes
date: Tue, 01 Mar 2022 02:53:44 GMT
via: 1.1 varnish
x-served-by: cache-iad-kiad7000058-IAD
x-cache: HIT
x-cache-hits: 3
x-timer: S1646103224.074590,VS0,VE0
vary: Authorization,Accept-Encoding,Origin
access-control-allow-origin: *
x-fastly-request-id: 06c20aa514f69eadca3884c0400f49592bad7094
expires: Tue, 01 Mar 2022 02:58:44 GMT
source-age: 80
content-length: 782

Reply:
No other glitches. Should work exactly as before when it was in-line. I thought about doing it in the header but wasn't sure if there was like an `<<__Override>>` for the header of a page in something like GuidePageController.

Reply:
We might as well put it in NonRoutableWebPageController if it's desired in every page except for error pages :)